Automattic/woocommerce-payments
# Fix fatal error on thank you page

Fixes #3411

#### Changes proposed in this Pull Request

Some customers have reported a fatal error where `get_currency` is called on a `bool` in the `init_order_currency` function. This happens if the return value of `wc_get_order` is `false` — this usually doesn't happen, but it's possible to reproduce it if an order is deleted and then the thank you page is viewed after that.

This PR fixes the issue by checking for a false order before calling `get_currency`. It also ensures that the order is always returned from the function in all branches of the filter.

#### Testing instructions

1. As a customer, place a new order. Once you get to the "Thank you" page for the order, save the URL of the thank you page so that you can re-visit it later.
2. As the shop manager, go to `/wp-admin/edit.php?post_type=shop_order` and move the newly placed order to the trash. Then go to the trash and delete the order permanently.
3. As the customer, refresh the "Thank you" page for the order. You should see a Thank you message with no fatal error (before, you would get a fatal error).

cc @jessepearson 

-------------------


**Post merge**
